### DISCUSSION TITLE
How to import WSL 2 Distro installed by another user?

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/WSL#6038] [General]
Hello,

My previous user profile which the WSL 2 distro was intalled is corrupted due to unfortunate events (and the NTUSER.DAT is inaccessible now). I followed the official guide and created a new user and copies everything around. Most of the things work, but WSL prompts:

> Windows Subsystem for Linux has no installed distributions. 
> Distributions can be installed by visiting the Windows Store: 
> https://aka.ms/wslstore
> 

I recognized that this might be registry data is not transferred along. I wonder what can I do to get the image registered.
For the record, it is a WSL 2 Distro (ubuntu), and version is 10.0.20226.1000.

Thank you.

**Comment by 0xbadfca11** [comment]
1. Install a new distro.
2. Search ``ext4.vhdx`` location. (e.g. ``REG.EXE QUERY HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss /s``.)
3. Exchange ``ext4.vhdx``.
#4591 #4762

Or, just recover files.
1. Mount ``ext4.vhdx`` in Windows. (Ignore unreadable filesystem)
2. [Mount that disk in WSL2.](https://docs.microsoft.com/en-us/windows/wsl/wsl2-mount-disk)

**Reply by craigloewen-msft** [reply]
I think recovering the files by mounting it into WSL 2 is a great idea using `wsl --mount`.
